daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/VariableMap1.scala

```diff
@@ -97,7 +98,8 @@ class VariableInstance private (
   val rd: VariableRuntimeData,
   @TransientParam defaultValueExprArg: => Maybe[CompiledExpression[AnyRef]],
   var priorState: VariableState = VariableUndefined,
-  var priorValue: DataValuePrimitiveNullable = DataValue.NoValue)
+  var priorValue: DataValuePrimitiveNullable = DataValue.NoValue,
```

Comment:
priorValue... I'm struggling with this. Why do we need that? When we create a NVI, the default value does not use that of an enclosing NVI. The DFDL spec says "The tuple is initialized based on the specifics of the dfdl:defineVariable annotation." The "tuple" being the state of a variable instance. So why do we need priorValue?

Reply:
Prior value is used in one place only right now.  When we are backtracking from a point of uncertainty where a variable was changed, when the PState is reset we also reset all the changed variables back to their previous state/value.

Reply:
Ok. Thanks